executor/executor.cc

```diff
@@ -548,6 +561,12 @@ void receive_execute()
 	}
 	if (req.prog_size == 0)
 		fail("need_prog: no program");
+#if GOOS_windows
+	DWORD dwRead = 0;
```

Comment:
Please add some abstraction layer for read/write functions. Throwing in more #ifdefs all over the place is not good.
The windows ones should go into executor_windows.h. The current impl, since it common for the rest of OSes, probably should go into executor.cc directly... or is there a better place?

Reply:
Okay we don't really need the DWORD and that can be ```NULL```. The only reason I did put in the ReadFile and WriteFile functions in execcutor_windows.h is becuase then I would still have to use ifdefs and since this is just a single line the advantage is less. 
```c
if (!ReadFile((HANDLE)_get_osfhandle(kInPipeFd), input_data, req.prog_size, NULL, NULL))
```

If you still want me to move them then do let me know.